testplan/common/utils/remote.py

```diff
@@ -135,3 +137,28 @@ def rm_cmd(path):
 def filepath_exist_cmd(path):
     """Checks if filepath exists."""
     return " ".join(["test", "-e", path])
+
+
+def calculate_pool_size_by_tasks(
+    tasks: Iterable[Task], plan_runtime_target: int
+):
+    _tasks = sorted(tasks, key=lambda task: task.weight, reverse=True)
+    max_weight = (
+        max(_tasks[0].weight, plan_runtime_target)
+        if _tasks
+        else plan_runtime_target
+    )
+    if max_weight > plan_runtime_target:
+        Warning(
+            f"Maximum task weight {max_weight} is greater than plan runtime target {plan_runtime_target}"
```

Comment:
Include the task object in the warning message.
And we might want to warn user about all tasks that has weight > plan_runtime_target if there are multiple of them